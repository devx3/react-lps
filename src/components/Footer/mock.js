export const mock = {
    html: 'Thanks little bastard. <a href="http://localhost">DevX3</a>.',
};
