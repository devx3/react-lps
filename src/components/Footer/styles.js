import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.div`
    text-align: center;
    ${({ theme }) => css`
        border-top: 1px solid ${theme.colors.mediumGrey};
        a {
            text-decoration: none;
        }

        & ${TextComponent} p {
            font-size: ${theme.fonts.sizes.xsmall};
        }

        & ${SectionContainer} {
            padding-top: 0;
            padding-bottom: 0;
        }
    `};
`;
