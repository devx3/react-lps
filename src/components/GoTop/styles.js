import styled, { css } from 'styled-components';

export const Container = styled.a`
    ${({ theme }) => css`
        background: ${theme.colors.secondaryColor};
        color: ${theme.colors.primaryColor};
        display: flex;
        align-items: center;
        justify-content: center;
        position: fixed;
        width: 4rem;
        height: 4rem;
        bottom: 2rem;
        right: 2rem;
        z-index: 6;
        cursor: pointer;
    `}
`;
