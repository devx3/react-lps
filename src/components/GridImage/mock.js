export const mock = {
    title: 'Hello GridImage',
    description: `Veniam velit magna nulla ex veniam sunt est.
    Consectetur ea aliquip do in. Qui amet non mollit elit proident
    aute adipisicing labore cillum. Non labore commodo minim labore
    ad sint eiusmod mollit anim cillum culpa officia cupidatat
    Lorem. Duis excepteur non ut consectetur adipisicing veniam sint.
    Aliquip ex excepteur non deserunt Lorem. Eu excepteur irure
    labore aliquip non aute sint sunt elit laboris occaecat ad veniam.`,
    grid: [
        {
            altText: 'Image 1',
            srcImage: 'https://source.unsplash.com/random/800x800?r=1',
        },
        {
            altText: 'Image 2',
            srcImage: 'https://source.unsplash.com/random/800x800?r=2',
        },
        {
            altText: 'Image 3',
            srcImage: 'https://source.unsplash.com/random/800x800?r=3',
        },
        {
            altText: 'Image 4',
            srcImage: 'https://source.unsplash.com/random/800x800?r=4',
        },
        {
            altText: 'Image 5',
            srcImage: 'https://source.unsplash.com/random/800x800?r=5',
        },
        {
            altText: 'Image 6',
            srcImage: 'https://source.unsplash.com/random/800x800?r=6',
        },
    ],
};
