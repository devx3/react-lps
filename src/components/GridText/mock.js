export const mock = {
    background: false,
    title: 'Um title qualquer',
    description: `Quis voluptate culpa dolor mollit minim non et.
    Veniam dolore amet exercitation non labore esse laborum. Aute
    nulla voluptate exercitation consectetur. Excepteur non et
    amet commodo irure. Eiusmod veniam laborum deserunt enim quis
    laborum cupidatat deserunt adipisicing id deserunt aute mollit
    culpa.`,
    grid: [
        {
            title: 'Title 1',
            description: `Quis voluptate culpa dolor mollit minim non et.
            Veniam dolore amet exercitation non labore esse laborum. Aute
            nulla voluptate exercitation consectetur. Excepteur non et
            amet commodo irure. Eiusmod veniam laborum deserunt enim quis
            laborum cupidatat deserunt adipisicing id deserunt aute mollit
            culpa.`,
        },
        {
            title: 'Title 2',
            description: `Quis voluptate culpa dolor mollit minim non et.
            Veniam dolore amet exercitation non labore esse laborum. Aute
            nulla voluptate exercitation consectetur. Excepteur non et
            amet commodo irure. Eiusmod veniam laborum deserunt enim quis
            laborum cupidatat deserunt adipisicing id deserunt aute mollit
            culpa.`,
        },
        {
            title: 'Title 3',
            description: `Quis voluptate culpa dolor mollit minim non et.
            Veniam dolore amet exercitation non labore esse laborum. Aute
            nulla voluptate exercitation consectetur. Excepteur non et
            amet commodo irure. Eiusmod veniam laborum deserunt enim quis
            laborum cupidatat deserunt adipisicing id deserunt aute mollit
            culpa.`,
        },
    ],
};
