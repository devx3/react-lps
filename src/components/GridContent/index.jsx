import * as Styled from './styles.js';
import P from 'prop-types';
import { SectionBackground } from '../SectionBackground';
import { Heading } from '../Heading';
import { TextComponent } from '../TextComponent';

const GridContent = ({ title, html, darkMode = false, sectionId = '' }) => {
    return (
        <SectionBackground background={darkMode} sectionId={sectionId}>
            <Styled.Container>
                <Heading as="h2" uppercase darkMode={darkMode}>
                    {title}
                </Heading>
                <Styled.Html>
                    <TextComponent>{html}</TextComponent>
                </Styled.Html>
            </Styled.Container>
        </SectionBackground>
    );
};

GridContent.propTypes = {
    title: P.string.isRequired,
    html: P.string.isRequired,
    darkMode: P.bool,
    sectionId: P.string,
};

export default GridContent;
