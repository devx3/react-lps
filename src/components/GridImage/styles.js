import styled, { css } from 'styled-components';
import { Container as TextContainer } from '../TextComponent/styles';

export const Container = styled.div`
    ${({ theme }) => css`
        ${TextContainer} {
            margin-bottom: ${theme.spacings.xhuge};
        }
    `}
`;

export const Grid = styled.div`
    ${({ theme }) => css`
        counter-reset: grid-counter;
        display: grid;
        grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
        gap: ${theme.spacings.large};
    `}
`;

export const GridItem = styled.div`
    ${() => css`
        overflow: hidden;
    `}
`;

export const Image = styled.img`
    ${() => css`
        width: 100%;
        transition: all 300ms ease-in-out;

        &:hover {
            transform: scale(1.2) rotate(5deg);
        }
    `}
`;
