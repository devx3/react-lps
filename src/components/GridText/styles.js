import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextContainer } from '../TextComponent/styles';

export const Container = styled.div`
    ${({ theme }) => css`
        ${TextContainer} {
            margin-bottom: ${theme.spacings.xhuge};
        }
    `}
`;

export const Grid = styled.div`
    ${({ theme }) => css`
        counter-reset: grid-counter;
        display: grid;
        grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
        gap: ${theme.spacings.large};

        @media ${theme.media.ltMedia} {
            grid-template-columns: 1fr;
        }
    `}
`;

export const GridItem = styled.div`
    ${({ theme }) => css`
        ${HeadingContainer} {
            position: relative;
            left: 5rem;
            margin-bottom: 15px;
        }

        ${HeadingContainer}::before {
            counter-increment: grid-counter;
            content: counter(grid-counter);
            position: absolute;
            top: -2rem;
            font-size: ${theme.fonts.sizes.huge};
            left: -5rem;
            transform: rotate(-10deg);
        }
    `}
`;
