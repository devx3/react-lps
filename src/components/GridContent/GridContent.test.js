import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import GridContent from '.';
import { mock } from './mock';

describe('<GridContent />', () => {
    it('Should render Grid Content', () => {
        const { container } = renderTheme(<GridContent {...mock} />);
        const heading = screen.getByRole('heading', {
            name: 'Hello GridContent',
        });

        expect(heading).toBeInTheDocument();
        expect(heading).toMatchSnapshot();
    });
});
