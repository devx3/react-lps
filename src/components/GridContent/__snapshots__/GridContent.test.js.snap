// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridContent /> Should render Grid Content 1`] = `
.c0 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<h2
  class="c0"
>
  Hello GridContent
</h2>
`;
